server/service/openapi/openapi.go

```diff
@@ -0,0 +1,288 @@
+	Summary     string `json:"summary,omitempty"`
+	Description string `json:"description,omitempty"`
+
+	GET     *Operation `json:"get,omitempty"`
+	PUT     *Operation `json:"put,omitempty"`
+	POST    *Operation `json:"post,omitempty"`
+	DELETE  *Operation `json:"delete,omitempty"`
+	OPTIONS *Operation `json:"options,omitempty"`
+	HEAD    *Operation `json:"head,omitempty"`
+	PATCH   *Operation `json:"patch,omitempty"`
+	TRACE   *Operation `json:"trace,omitempty"`
+
+	Servers    []*Server    `json:"servers,omitempty"`
+	Parameters []*Parameter `json:"parameters,omitempty"`
+}
+
+type Parameter struct {
+	Name        string `json:"name"`
+	In          string `json:"in"` // must be "query", "header", "path" or "cookie"
+	Description string `json:"description,omitempty"`
+	Required    bool   `json:"required"`
+	Deprecated  bool   `json:"deprecated,omitempty"`
+	// AllowEmptyValue (use is NOT RECOMMENDED, likely to be removed from spec)
+
+	// next are variable fields - used to specify serialization of the parameter
+
+	// Style
+	// Explode
+	// AllowReserved
+	Schema *Schema `json:"schema,omitempty"`
+	// Example | Examples
+	// Content
+}
+
+type Operation struct {
+	Tags        []string `json:"tags,omitempty"`
+	Summary     string   `json:"summary,omitempty"`
+	Description string   `json:"description,omitempty"`
+	// ExternalDocs
+	OperationID string            `json:"operationId,omitempty"`
+	Parameters  []*Parameter      `json:"parameters,omitempty"`
+	RequestBody *RequestBody      `json:"requestBody,omitempty"`
+	Responses   ResponsesPatterns `json:"responses,omitempty"`
+	// Callbacks
+	Deprecated bool `json:"deprecated,omitempty"`
+	// Security
+	Servers []*Server `json:"servers,omitempty"`
+}
+
+type RequestBody struct {
+	Description string               `json:"description,omitempty"`
+	Content     map[string]MediaType `json:"content"`
+	Required    bool                 `json:"required"`
+}
+
+type ResponsesPatterns []*ResponsePattern
+
+func (p ResponsesPatterns) MarshalJSON() ([]byte, error) {
+	m := make(map[string]Response, len(p))
+	for _, pat := range p {
+		m[pat.StatusCode] = pat.Response
+	}
+	return json.Marshal(m)
+}
+
+type ResponsePattern struct {
+	StatusCode string // can be "default" or an HTTP status code
+	Response   Response
+}
+
+func (p ResponsePattern) MarshalJSON() ([]byte, error) {
+	return json.Marshal(map[string]Response{p.StatusCode: p.Response})
+}
+
+type Response struct {
+	Description string               `json:"description"`
+	Headers     map[string]Parameter `json:"headers,omitempty"` // Same as Parameter, but no Name (comes from map) nor In (implicit).
+	Content     map[string]MediaType `json:"content,omitempty"`
+	// Links
+}
+
+type MediaType struct {
+	Schema Schema `json:"schema"`
+	// Example | Examples
+	// Encoding
+}
+
+type Schema struct {
+	// This object is an extended subset of the JSON Schema Specification Wright
+	// Draft 00 (https://json-schema.org/).
+	jsonschema.Schema
```

Comment:
Given that openapi schema is almost exactly the jsonschema, there's a quite complete existing package that supports this and is used here instead of trying to reimplement this from scratch (jsonschema is relatively complex).